tardis/visualization/tools/liv_plot.py

```diff
@@ -199,21 +199,13 @@ def _generate_plot_data(self, packets_mode):
         ----------
         packets_mode : str
             Packet mode, either 'virtual' or 'real'.
-
-        Returns
-        -------
-        plot_data : list
-            List of velocity data for each species.
-
-        plot_colors : list
-            List of colors corresponding to each species.
         """
         groups = self.data[packets_mode].packets_df_line_interaction.groupby(
             by="last_line_interaction_species"
         )
 
-        plot_colors = []
-        plot_data = []
+        self.plot_colors = []
+        self.plot_data = []
```

Comment:
plot_data and plot_colors are being used by multiple methods so I thought it would be better to define them as instance variables instead of calling the function and assigning them to a local variable where ever we need them.